core/pom.xml

```diff
@@ -275,6 +275,10 @@
               <excludes />
               <allows>
                 <allow>io[.]opentelemetry[.]api[.]OpenTelemetry</allow>
+                <allow>org[.]apache[.]hadoop[.]conf[.]Configuration</allow>
+                <allow>org[.]apache[.]hadoop[.]io[.]compress[.]CompressionCodec</allow>
+                <allow>org[.]apache[.]hadoop[.]io[.]compress[.]Decompressor</allow>
+                <allow>org[.]apache[.]hadoop[.]io[.]compress[.]Compressor</allow>
```

Comment:
Rather than allowing these types to be exposed in our SPI, why not refactor to avoid exposing the Hadoop types? Do we really need these types? Can we use our own analogous ones? Or can we avoid them entirely, deferring to the plugin implementor?

Reply:
> Is that what you are suggesting, that we move the internal implementation away from Hadoop Codecs?

We could, but that's not necessarily what I'm suggesting. I'm just suggesting that it may be possible to use Accumulo-only types in the SPI, and then wrap the or unwrap the configured implementing class to present it as a Hadoop Codec. That way, we're not bound in our SPI to Hadoop types if we want to change the internal implementation later.

Reply:
I think the probability of moving from Hadoop Codecs to something else is very low. It would likely occur at the same time that we implement our own FileSystem layer, which would likely have to happen in a major release where we could change the public API anyway.

Reply:
You could just create our own type for SPI and make all the Hadoop types internal private variables

Reply:
My intent in proposing this is to avoid introducing *new* exposures of Hadoop types in our API/SPI. Even if we're not moving away from it soon, the more we expose in our API/SPI, the harder it is to abstract Hadoop out later.

Reply:
> You could just create our own type for SPI and make all the Hadoop types internal private variables

It's not clear to me how a plug-in would create an object and then we convert it to a Codec / Compressor / Decompressor in the internal code. Do you have an example where this is done elsewhere?

Reply:
If the methods on our own plugin provided all the necessary functionality, it would be trivial to write a pass-through Hadoop Codec that merely called our SPI methods. Then, whatever the user configures can be easily wrapped by that single Codec internally. For implementing classes we provide, we could even make all our SPI methods return `UnsupportedOperationException`, and instead just use its name, as in:

```java
Codec getCodec() {
  var configuredCompressionPlugin = instantiateConfiguredClass();
  if (configuredCompressionPlugin.getClass().equals(AccumuloGZipCompression.class)) {
    return new GZipCodec(); // return Hadoop's equivalent
  } else if (
    // check other built-in known types
  ) {
    // ....
  } else {
    return new InternalCodecWrapperForAccumuloCompressionPlugin(configuredCompressionPlugin);
  }
}
```

Reply:
I backed out the changes to core/pom.xml in 91942f7